### PR TITLE
Add DB connection check, example endpoints, and docker compose

### PR DESCRIPTION
**Pull request by LeavesOfFall**
- Check that the DB is accessible on init
- example endpoints `/read/:id` and `/write` for reading and writing to an example collection; to be removed once we have actual endpoints using the DB
- Docker compose for running DB for local testing/development

### CODE REVIEW

File: src/app/mod.rs [hunk 1]
```diff
@@ -1,32 +1,48 @@
...
+    /// Will return 'Err' if the app cannot connect to the database
     pub async fn new() -> anyhow::Result<Self> {
         let database_url = config::get_database_url();
         let client = Client::with_uri_str(database_url).await?;
         let db = client.database("capstone");
-        Ok(State { db: Arc::new(db) })
+        // Ping DB to make sure we can connect
+        db.run_command(doc! { "ping": 1 })
+            .await
+            .inspect_err(|e| tracing::error!(error = %e, "Failed to connect to database"))?;
```
Comment (Jack17432) [inline src/app/mod.rs:37]: add context to error when pass back

File: src/config.rs [hunk 2]
```diff
@@ -1,7 +1,7 @@
 use std::net::{IpAddr, Ipv4Addr, SocketAddr};
 
 const DEFAULT_APT_PORT: u16 = 4444;
-const DEFAULT_DATABASE_URL: &str = "mongodb://localhost:27107";
+const DEFAULT_DATABASE_URL: &str = "mongodb://localhost:27017";
```
Comment (Jack17432) [inline src/config.rs:4]: Make this an env and have a retrival funx

File: src/app/mod.rs [hunk 3]
```diff
@@ -112,3 +130,51 @@ fn foo() {
...
+    number: i32,
+}
+
+#[tracing::instrument]
+async fn read_example(Path(id): Path<String>, State(db): State<Database>) -> Response {
+    // TODO: Remove example after first endpoint made
+    let document = db
+        .collection::<ExampleDocument>("testCollection")
+        .find_one(doc! { "_id": id })
+        .await;
```
Comment (Jack17432) [inline src/app/mod.rs:148]: qualify ID as int so that the db can auto increment id

File: src/app/mod.rs [hunk 4]
```diff
@@ -112,3 +130,51 @@ fn foo() {
     // TODO: Remove example after first endpoint made
     tracing::warn!("foo!");
 }
+
+#[derive(Serialize, Deserialize, Debug)]
+struct ExampleDocument {
+    #[serde(rename = "_id")]
+    id: String,
```
Comment (Jack17432) [inline src/app/mod.rs:137]: Int

**Review by Jack17432 — CHANGES_REQUESTED** [review]
Small changes

File: src/config.rs (hunk 2 above)
Comment (LeavesOfFall) [reply src/config.rs:4]: This is just the fallback when there's no specified URL, we already have `DATABASE_URL` as an env variable

Comment (LeavesOfFall) [reply src/config.rs:4]: (27017 is the default port, pretty sure 27107 was just a typo)

**Review by Jack17432 — APPROVED** [review]
looks good as an example

**Pull request merged**